app/components/enrolment_confirmation_component/enrolment_confirmation_component.html.erb

```diff
@@ -0,0 +1,32 @@
+<% if @current_user %>
+  <% if @programme.enrolment_confirmation_required? %>
+    <div class="enrolment-confirmation-component">
+      <%= render ModalComponent.new(
+        title: "Enrol on the #{@programme.certificate_name}",
+        reopen_button_text: button_text,
+        show_corner_decoration: false,
+        reopen_button_class: ("enrolment-confirmation-component__button--full-width" if @full_width)
+        ) do |component| %>
+        <% component.with_body do %>
+          <div class="enrolment-confirmation-component__wrapper">
+            <div>
+              <h2 class="govuk-heading-s">Ready to enrol?</h2>
+              <p class="govuk-body">Confirm your enrolment on the certificate, and then let's find you the perfect training course to get started with.</p>
+            </div>
+            <div>
+              <%= link_to "Confirm my enrolment", @programme.enrol_path(user_programme_enrolment: { user_id: @current_user.id, programme_id: @programme.id }), method: :post, class: "govuk-button button", role: "button" %>
+            </div>
+          </div>
+        <% end %>
+      <% end %>
+    </div>
+  <% else %>
+    <% if @programme.user_enrolled?(@current_user) %>
```

Comment:
Is the logic here right?

What happens if I am logged in, and enrolled on a programme that requires enrolment confirmation? Should the check for if they enrolled be before checking programme type?

Reply:
Yes, good spot. The enrolment state needs to be checked before the programme.